src/components/siteEntryTable/index.tsx

```diff
@@ -1,42 +1,129 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import {
   SiteEntry,
   SiteEntryField,
   SiteEntryFields,
 } from '../../containers/treePage/ducks/types';
-import { Table } from 'antd';
+import { Form, Modal, Table, message } from 'antd';
+import { EditOutlined } from '@ant-design/icons';
+import ProtectedClient from '../../api/protectedApiClient';
 import {
   booleanToString,
   getSEFieldDisplayName,
+  n,
 } from '../../utils/stringFormat';
+import { EditButton, StyledClose } from '../careEntry';
+import UpdateSiteForm from '../forms/updateSiteForm';
+import { SiteEntriesRequest, UpdateSiteRequest } from '../forms/ducks/types';
+import { useTranslation } from 'react-i18next';
+import { site } from '../../constants';
+import moment from 'moment';
 
 interface SiteEntryTableProps {
   readonly siteEntries: SiteEntry[];
+  readonly getSite: () => void;
 }
 
-const Columns = Object.values(SiteEntryFields).map((field: SiteEntryField) => {
-  return {
-    title: getSEFieldDisplayName(field),
-    dataIndex: field,
-    key: field,
-    render: (val: string | number | boolean): string => {
-      if (val || val === false) {
-        return booleanToString(String(val));
-      } else {
-        return '';
-      }
+const SiteEntryTable: React.FC<SiteEntryTableProps> = ({
+  siteEntries,
+  getSite,
+}) => {
+  const { t } = useTranslation(n(site, ['site', 'forms']), {
+    nsMode: 'fallback',
+  });
+
+  const [showEditEntryModal, setShowEditEntryModal] = useState<boolean>(false);
+  const [editEntryModalData, setEditEntryModalData] = useState<SiteEntry>();
+
+  const [editSiteEntryForm] = Form.useForm();
+
+  useEffect(() => {
+    editSiteEntryForm.setFieldsValue({
+      ...editEntryModalData,
+      plantingDate: editEntryModalData?.plantingDate
+        ? moment(editEntryModalData?.plantingDate)
+        : null,
+    });
+  }, [editSiteEntryForm, editEntryModalData]);
```

Comment:
⛏️ It looks like we only change the modal data in the click function of each row's `EditButton` - can we move this logic into that function instead of a `useEffect`, or is there an edge case where this is needed?

Reply:
Ah took a look again and I can move it. Nice catch!